overlord/devicestate/devicemgr.go

```diff
@@ -1041,6 +1041,57 @@ func (m *DeviceManager) ensureAutoImportAssertions() error {
 	return nil
 }
 
+func (m *DeviceManager) ensureSerialBoundAssertionsProcessed() error {
+	if release.OnClassic {
+		return nil
+	}
+
+	if m.earlyDeviceSeed != nil {
+		return nil
+	}
+
+	m.state.Lock()
+	defer m.state.Unlock()
+
+	var seeded bool
+	if err := m.state.Get("seeded", &seeded); err != nil && !errors.Is(err, state.ErrNoState) {
+		return err
+	}
+	if !seeded {
+		return nil
+	}
+
+	var waitingOnSerial bool
+	err := m.state.Get("assertion-waiting-on-serial", &waitingOnSerial)
+	if err != nil && !errors.Is(err, state.ErrNoState) {
+		logger.Noticef("failed to get serial-bound assert state")
+	}
+	if !waitingOnSerial {
+		return nil
+	}
+
+	model, err := m.Model()
+	if err != nil {
+		return err
+	}
+
+	serial, err := m.Serial()
+	if err != nil {
+		return nil
+	}
+
+	db := assertstate.DB(m.state)
+	m.state.Set("assertion-waiting-on-serial", false)
```

Comment:
Potentially this should be set *after* we know `createAllKnownSystemUsers` didn't fail?